lib/remote/jsonrpcconnection-pki.cpp

```diff
@@ -207,23 +196,11 @@ Value RequestCertificateHandler(const MessageOrigin::Ptr& origin, const Dictiona
 		}
 	}
 
-	pubkey = std::shared_ptr<EVP_PKEY>(X509_get_pubkey(cert.get()), EVP_PKEY_free);
-	subject = X509_get_subject_name(cert.get());
+	newcert = cert;
```

Comment:
And if `RenewCert` just returned a new pointer, this could just be `std::shared_ptr<X509> newcert = listener->RenewCert(newcert);` (keeping the variable declaration way up there doesn't make sense to me).